---
title: "class09 structural bioinformatics"
author: "Jacob Gil"
format: pdf
---

```{r}
pdbdata <- read.csv("Data Export Summary.csv")
pdbdata
```
>Q1: What percentage of structures in the PDB are solved by X-Ray and Electron Microscopy.

93.15%

>Q2: What proportion of structures in the PDB are protein?

86.67%

>Q3: Type HIV in the PDB website search box on the home page and determine how many HIV-1 protease structures are in the current PDB?

200

>Q4: Water molecules normally have 3 atoms. Why do we see just one atom per water molecule in this structure?

Only the hydrogen bonding hydrogens in the water molecule are shown

>Q5: There is a critical “conserved” water molecule in the binding site. Can you identify this water molecule? What residue number does this water molecule have

Asp 25 binds the water molecule

>Q6: Generate and save a figure clearly showing the two distinct chains of HIV-protease along with the ligand. You might also consider showing the catalytic residues ASP 25 in each chain and the critical water (we recommend “Ball & Stick” for these side-chains). Add this figure to your Quarto document.
![1HSG](1HSG.png)
>Q7: [Optional] As you have hopefully observed HIV protease is a homodimer (i.e. it is composed of two identical chains). With the aid of the graphic display can you identify secondary structure elements that are likely to only form in the dimer rather than the monomer?

There is a beta sheet that forms between the two monomers that would not be possible if there was only one monomer.

```{r}
library(bio3d)
pdb <- read.pdb("1hsg")
pdb
```
>Q7: How many amino acid residues are there in this pdb object? 

198

>Q8: Name one of the two non-protein residues? 

MK1

>Q9: How many protein chains are in this structure?

2

```{r}
attributes(pdb)
```
```{r}
head(pdb$atom)
```
```{r}
adk <- read.pdb("6s36")
```
```{r}
adk
```
```{r}
# Perform flexiblity prediction
m <- nma(adk)
```
```{r}
plot(m)
```
```{r}
mktrj(m, file="adk_m7.pdb")
```

>Q10. Which of the packages above is found only on BioConductor and not CRAN? 

msa

>Q11. Which of the above packages is not found on BioConductor or CRAN?: 

Grantlab/bio3d-view

>Q12. True or False? Functions from the devtools package can be used to install packages from GitHub and BitBucket? 

True

```{r}
library(bio3d)
aa <- get.seq("1ake_A")
```
```{r}
aa
```
>Q13. How many amino acids are in this sequence, i.e. how long is this sequence?

214

```{r}
# Blast or hmmer search 
#b <- blast.pdb(aa)
```
```{r}
# Plot a summary of search results
#hits <- plot(b)
hits <- NULL
hits$pdb.id <- c('1AKE_A','6S36_A','6RZE_A','3HPR_A','1E4V_A','5EJE_A','1E4Y_A','3X2S_A','6HAP_A','6HAM_A','4K46_A','3GMT_A','4PZL_A')

```
```{r}
# List out some 'top hits'
head(hits$pdb.id)
```
```{r}
# Download releated PDB files
files <- get.pdb(hits$pdb.id, path="pdbs", split=TRUE, gzip=TRUE)
```
```{r}
# Align releated PDBs
pdbs <- pdbaln(files, fit = TRUE)
```
```{r}
# Vector containing PDB codes for figure axis
ids <- basename.pdb(pdbs$id)

# Draw schematic alignment
#plot(pdbs, labels=ids)
# Receieved this when rendering to PDF: 
# Error in `plot.new()`:
# ! figure margins too large
```
```{r}
anno <- pdb.annotate(ids)
unique(anno$source)
```
```{r}
anno
```
```{r}
# Perform PCA
pc.xray <- pca(pdbs)
plot(pc.xray)
```

```{r}
# Calculate RMSD
rd <- rmsd(pdbs)

# Structure-based clustering
hc.rd <- hclust(dist(rd))
grps.rd <- cutree(hc.rd, k=3)

plot(pc.xray, 1:2, col="grey50", bg=grps.rd, pch=21, cex=1)
```
```{r}
# Visualize first principal component
pc1 <- mktrj(pc.xray, pc=1, file="pc_1.pdb")
```
```{r}
#Plotting results with ggplot2
library(ggplot2)
library(ggrepel)

df <- data.frame(PC1=pc.xray$z[,1], 
                 PC2=pc.xray$z[,2], 
                 col=as.factor(grps.rd),
                 ids=ids)

p <- ggplot(df) + 
  aes(PC1, PC2, col=col, label=ids) +
  geom_point(size=2) +
  geom_text_repel(max.overlaps = 20) +
  theme(legend.position = "none")
p
```
```{r}
# NMA of all structures
modes <- nma(pdbs)
plot(modes, pdbs, col=grps.rd)
```
>Q14. What do you note about this plot? Are the black and colored lines similar or different? Where do you think they differ most and why?

The black lines have less fluctuation than the colored lines. They likely differ most in parts of the protein that are not folded into secondary structures such as sheets or heliecies. The active sites are also likely places with low fluctuation.

